components/inputs/input-date.js

```diff
@@ -503,10 +516,13 @@ class InputDate extends LabelledMixin(SkeletonMixin(FormElementMixin(LocalizeCor
 		this._formattedValue = this._shownValue ? formatISODateInUserCalDescriptor(this._shownValue) : (this.emptyText ? this.emptyText : '');
 	}
 
-	async _updateValueDispatchEvent(dateInISO) {
-		if (dateInISO === this._shownValue) return; // prevent validation from happening multiple times for same change
+	async _updateValueDispatchEvent(dateInISO, setToNow) {
+		// prevent validation from happening multiple times for same change,
+		// except for now button that affects time
+		if (!setToNow && dateInISO === this._shownValue) return;
 		this._shownValue = dateInISO;
 		this.value = dateInISO;
+		this.setToNow = setToNow;
```

Comment:
Instead of storing this on the element (which becomes meaningless after the event is dispatched), I'd suggest storing it in the event detail (an optional 3rd parameter on `CustomEvent`). Then the listener can check that instead of looking at the target.